jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/GradleLayerConfigurations.java

```diff
@@ -58,45 +63,86 @@ static JavaLayerConfigurations getForProject(
     List<Path> resourcesFiles = new ArrayList<>();
     List<Path> classesFiles = new ArrayList<>();
     List<Path> extraFiles = new ArrayList<>();
+    final boolean isWebapp = GradleProjectProperties.getWarTask(project) != null;
 
-    // Adds each file in each classes output directory to the classes files list.
-    FileCollection classesOutputDirectories = mainSourceSet.getOutput().getClassesDirs();
-    gradleJibLogger.info("Adding corresponding output directories of source sets to image");
-    for (File classesOutputDirectory : classesOutputDirectories) {
-      if (Files.notExists(classesOutputDirectory.toPath())) {
-        gradleJibLogger.info("\t'" + classesOutputDirectory + "' (not found, skipped)");
-        continue;
+    if (isWebapp) {
```

Comment:
This method became too long, so let's break it down. Some notes:

- No `final` on local variables (`isWebapp`).
- `isWebapp` --> `isWebApp`. Perhaps `isWarProject`.

```java
  static JavaLayerConfigurations getForProject(
      Project project, GradleJibLogger logger, Path extraDirectory, String appRoot)
      throws IOException {
    boolean isWarProject = GradleProjectProperties.getWarTask(project) != null;
    if (isWarProject) {
      logger.info("WAR project identified, creating WAR image: " + project.getDisplayName());
      return getForWarProject(war, logger, extraDirectory, appRoot);
    } else {
      return getForJarProject(project, logger, extraDirectory, appRoot);
    }
  }
```
where `getForJarProject()` is the previous `getForProject()` (I don't think you will touch anything on it).

Reply:
Done